cmd/kubeadm/app/componentconfigs/registrations.go

```diff
@@ -60,13 +64,20 @@ func (r Registration) Unmarshal(fileContent []byte) (runtime.Object, error) {
 	obj := r.EmptyValue.DeepCopyObject()
 
 	// Decode the file content into obj which is a pointer to an empty struct of the internal ComponentConfig
-	// object, using the componentconfig Codecs that knows about all APIs
-	if err := runtime.DecodeInto(Codecs.UniversalDecoder(), fileContent, obj); err != nil {
+	if err := unmarshalObject(obj, fileContent); err != nil {
 		return nil, err
 	}
 	return obj, nil
 }
 
+func unmarshalObject(obj runtime.Object, fileContent []byte) error {
```

Comment:
I swear these functions exist elsewhere.

Reply:
This function uses a scheme with component configs api registered.
The other `unmarshal` func uses a scheme with only kubeadm types registered